src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,212 @@
+     * @param {string} oldPassword
+     */
+    async _unlock(oldPassword) {
+        TopLevelApi.setLoading(true);
+        const oldPasswordBytes = oldPassword ? Utf8Tools.stringToUtf8ByteArray(oldPassword) : undefined;
+        /** @type {Key?} */
+        let key = null;
+        try {
+            key = await KeyStore.instance.get(this._request.keyInfo.id, oldPasswordBytes);
+        } catch (e) {
+            if (e.message === 'Invalid key') {
+                TopLevelApi.setLoading(false);
+                this._passwordGetter.onPassphraseIncorrect();
+                return;
+            }
+            this._reject(new Errors.CoreError(e));
+            return;
+        }
+        if (!key) {
+            this._reject(new Errors.KeyNotFoundError());
+            return;
+        }
+        this._key = key;
+        this._passwordSetter.reset();
+        window.location.hash = ChangePassword.Pages.SET_PASSWORD;
+        if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
+            this._passwordSetter.focus();
+        }
+        TopLevelApi.setLoading(false);
+    }
+
+
+    /**
+     * Called after new password was entered first time.
+     */
+    async _prepare() {
+        let colorClass = '';
+        if (this.key.secret instanceof Nimiq.Entropy) {
+            const color = Iqons.getBackgroundColorIndex(
+                new Nimiq.Address(
+                    // use color of first address as loginFile color
+                    this.key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                ).toUserFriendlyAddress(),
+            );
+            const colorString = LoginFile.CONFIG[color].name;
+            colorClass = `nq-${colorString}-bg`;
+        }
+        this._loginFileIcon.lock(colorClass);
+        this._progressIndicator.setStep(3);
+    }
+
+    /**
+     * Called after new password was entered second time.
+     * @param {string} newPassword
+     */
+    async _commitChangeAndOfferLoginFile(newPassword) {
+        TopLevelApi.setLoading(true);
+        const passwordBytes = Utf8Tools.stringToUtf8ByteArray(newPassword);
+
+        await KeyStore.instance.put(this.key, passwordBytes);
+
+        if (this.key.secret instanceof Nimiq.PrivateKey) {
+            // Login File not available for legacy accounts
+            this._resolve({ success: true });
+            return;
+        }
+
+        // Prepare Login File for download
+
+        const firstAddress = new Nimiq.Address(
+            this.key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+        );
+
+        const encryptedEntropy = await this.key.secret.exportEncrypted(passwordBytes);
+
+        this._downloadLoginFile.setEncryptedEntropy(
+            /** @type {Nimiq.SerialBuffer} */(encryptedEntropy),
+            firstAddress,
+        );
+
+        this._downloadLoginFile.on(DownloadLoginFile.Events.DOWNLOADED, () => {
```

Comment:
what happens if I go back and fourth between the login file and the password?

I think the event handling here should go to the constructor.